File: programmers/level1/신고_결과_받기.rb
def solution(id_list, report, k)
    answer = []
    reported_list = {}
    mail_list = {}
    
    id_list.each{ |id|
        reported_list[id] = []
        mail_list[id] = 0
    }
    
    report.each{ |re|
        reporter, user = re.split " "
        if !reported_list[user].include?(reporter)
            reported_list[user] << reporter
        end
    }
    
    id_list.each{ |id|
        if reported_list[id].length >= k
            reported_list[id].each{ |mail_id|
                mail_list[mail_id] += 1
            }
        end
    }
    
    id_list.each{ |key|
        answer << mail_list[key]
    }

    return answer
end

File: programmers/level1/신고_결과_받기.py
def solution(id_list, report, k):
    answer = [0]*len(id_list)
    ind={}
    for i in id_list:
        ind[i]=set()
    for i in set(report):
        report_id,reported_id=i.split()
        ind[reported_id].add(report_id)

    for i in ind.keys():
        if len(ind[i])>=k:
            for j in ind[i]:
               answer[id_list.index(j)]+=1
    return answer